[DeviceMesh] Rename get_dim_groups to get_group

wz337 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #114710
* #114709
* __->__ #114708

Rename get_dim_groups to get_group and update all callsites.

Differential Revision: [D51629801](https://our.internmc.facebook.com/intern/diff/D51629801/)

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @tianyu-l @wconstab @yf225 @kiukchung @d4l3k @lucasllc

wz337 committed:
[DeviceMesh] Rename get_dim_groups to get_group

Rename get_dim_groups to get_group and update all callsites.

Differential Revision: [D51629801](https://our.internmc.facebook.com/intern/diff/D51629801/)

[ghstack-poisoned]

wanchaol commented on torch/distributed/_device_mesh.py:
```diff
@@ -339,13 +339,28 @@ def __getitem__(self, mesh_dim_name: str) -> "DeviceMesh":
     ) -> Union[ProcessGroup, List[ProcessGroup]]:
+        """
+        Returns a list of ProcessGroups corresponding to the mesh dimensions, or
+        returns a single ProcessGroup if mesh_dim is specified or a given mesh has
```
```suggestion
        returns a single ProcessGroup if mesh_dim is specified or the given mesh has
```

XilunWu reviewed (approved):
LGTM! I would prefer "get_group_by_dim" or "get_dim_group" but it's just personal.

wanchaol commented on torch/distributed/_device_mesh.py:
```diff
@@ -339,13 +339,28 @@ def __getitem__(self, mesh_dim_name: str) -> "DeviceMesh":
             raise RuntimeError("DeviceMesh process groups not initialized!")
         if mesh_dim is not None:
+            if isinstance(mesh_dim, str):
+                mesh_dim = self.mesh_dim_names.index(mesh_dim)
```
nit: add assertion here in case `self.mesh_dim_names` is None or empty

wanchaol commented on torch/distributed/_device_mesh.py:
```diff
@@ -339,13 +339,28 @@ def __getitem__(self, mesh_dim_name: str) -> "DeviceMesh":
+            if isinstance(mesh_dim, str):
+                mesh_dim = self.mesh_dim_names.index(mesh_dim)
             return _find_pg_by_ranks_and_tag(*self._dim_group_infos[mesh_dim])
+        elif self.mesh.ndim == 1:
```
nit: this can be checked in the beginning before `if mesh_dim is not None:`?

wanchaol reviewed (approved):
Only looked at the API change haven't looked into other changes, assuming that it's mostly a test refactor.

cc @awgu `get_dim_groups()` API name changing to `get_group` for 2.2 release, in case you need to adjust your code

awgu commented on torch/distributed/_device_mesh.py:
```diff
@@ -339,13 +339,28 @@ def __getitem__(self, mesh_dim_name: str) -> "DeviceMesh":
+            mesh_dim (str/int): it can be the name of the mesh dimension or the index
+            of the mesh dimension. Default is None.
+        Returns:
+            A list of :class:`ProcessGroup` object or a single :class:`ProcessGroup` object.
```
nit: Let us explain when we expect which type?

wz337 committed:
Update on "[DeviceMesh] Rename get_dim_groups to get_group"


Rename get_dim_groups to get_group and update all callsites.

Differential Revision: [D51629801](https://our.internmc.facebook.com/intern/diff/D51629801/)

cc H-Huang awgu kwen2501 wanchaol fegin fduwjj wconstab mrshenli zhaojuanmao rohan-varma kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]

wz337 committed:
Update on "[DeviceMesh] Rename get_dim_groups to get_group"


Rename get_dim_groups to get_group and update all callsites.

Differential Revision: [D51629801](https://our.internmc.facebook.com/intern/diff/D51629801/)

cc H-Huang awgu kwen2501 wanchaol fegin fduwjj wconstab mrshenli zhaojuanmao rohan-varma kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]

wz337 committed:
Update on "[DeviceMesh] Rename get_dim_groups to get_group"


Rename get_dim_groups to get_group and update all callsites.

Differential Revision: [D51629801](https://our.internmc.facebook.com/intern/diff/D51629801/)

cc H-Huang awgu kwen2501 wanchaol fegin fduwjj wconstab mrshenli zhaojuanmao rohan-varma kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]

fegin reviewed (approved):
There is a distributed test failure. Can we check if it is related?

wz337 committed:
Update on "[DeviceMesh] Rename get_dim_groups to get_group"


Rename get_dim_groups to get_group and update all callsites.

Differential Revision: [D51629801](https://our.internmc.facebook.com/intern/diff/D51629801/)

cc H-Huang awgu kwen2501 wanchaol fegin fduwjj wconstab mrshenli zhaojuanmao rohan-varma kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]

wz337 committed:
Update on "[DeviceMesh] Rename get_dim_groups to get_group"


Rename get_dim_groups to get_group and update all callsites.

Differential Revision: [D51629801](https://our.internmc.facebook.com/intern/diff/D51629801/)

cc H-Huang awgu kwen2501 wanchaol fegin fduwjj wconstab mrshenli zhaojuanmao rohan-varma kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]

wz337 committed:
Update on "[DeviceMesh] Rename get_dim_groups to get_group"


Rename get_dim_groups to get_group and update all callsites.

Differential Revision: [D51629801](https://our.internmc.facebook.com/intern/diff/D51629801/)

cc H-Huang awgu kwen2501 wanchaol fegin fduwjj wconstab mrshenli zhaojuanmao rohan-varma kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]

wz337 committed:
Update on "[DeviceMesh] Rename get_dim_groups to get_group"


Rename get_dim_groups to get_group and update all callsites.

Differential Revision: [D51629801](https://our.internmc.facebook.com/intern/diff/D51629801/)

cc H-Huang awgu kwen2501 wanchaol fegin fduwjj wconstab mrshenli zhaojuanmao rohan-varma kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]
